### PR TITLE
Add learning rate decay and exponential decay schedule

### PR DESCRIPTION
**Pull request by nsorros**
Description
---

Adds learning rate decay parameter which allows for a more sophisticated learning rate schedule. At the moment the way this works it that a decay is multiplied with the learning rate at the end of each epoch.

Checklist
---

- [x] Added link to Github issue or Trello card
- [x] Added tests

### CODE REVIEW

**Comment by nsorros** [comment]
[Add learning rate schedule ability to CNN and BiLSTM](https://trello.com/c/rPfQuOvd/1122-add-learning-rate-schedule-ability-to-cnn-and-bilstm)

**Comment by nsorros** [comment]
@aCampello @lizgzil

**Comment by nsorros** [comment]
@aCampello @lizgzil any thoughts on this?

**Review by lizgzil — APPROVED** [review]
looks good to me. So setting `learning_rate_decay=1` means that there is no decay I take it?

**Comment by nsorros** [comment]
> looks good to me. So setting `learning_rate_decay=1` means that there is no decay I take it?

yes, for backwards compatibility
